unverified/sexpr-bootstrap/x64/64/extract_code.cml

```diff
@@ -0,0 +1,48 @@
+(*
+  A CakeML program that extracts the indented code snippets from a
+  markdown file. This is used for checking the code in how-to.md.
+*)
+
+val lines =
+  let
+    val arg = List.hd (CommandLine.arguments())
+  in
+    Option.valOf (TextIO.inputLinesFrom arg)
+  end
+  handle _ =>
+    (TextIO.print_err ("usage: " ^ CommandLine.name() ^ " <filename>\n");
+     raise Bind);
+
+fun takeUntil p xs =
```

Comment:
These aren't in the basis library? If not, they should be. (Can be separate commits later.)

Reply:
By "these", do you mean `takeUntil` and `dropUntil`?

Reply:
Yes